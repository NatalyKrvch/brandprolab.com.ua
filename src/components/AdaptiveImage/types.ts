export interface PictureSource {
  srcSet: string;
  media: string;
}

export interface AdaptiveImageProps {
  sources: PictureSource[];
  fallbackSrc: string | null;
  alt: string;
  className?: string;
}
