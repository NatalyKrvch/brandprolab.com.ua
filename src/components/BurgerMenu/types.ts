import type { MenuItemType } from '../NavigationMenu';
import type { SocialIcon } from '../SocialIcons';

export interface BurgerMenuProps {
  isOpen: boolean;
  onClose: () => void;
  photoUrl: string;
  socialLinks: SocialIcon[];
  menuItems: MenuItemType[];
  panelClassName?: string;
  navClassName?: string;
  animationMs?: number;
  portalId?: string;
}
