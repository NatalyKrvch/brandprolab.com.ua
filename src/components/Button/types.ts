export type ButtonColor = 'teal' | 'white';

export interface ButtonProps {
  onClick?: () => void;
  color?: ButtonColor;
}
