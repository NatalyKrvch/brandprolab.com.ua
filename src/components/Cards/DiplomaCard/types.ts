export interface DiplomaCardProps {
  diplomaURL: string;
  onClick?: () => void;
}
