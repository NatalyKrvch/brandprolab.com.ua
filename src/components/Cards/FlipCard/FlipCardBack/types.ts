export interface FlipCardBackProps {
  header: string;
  list: string[];
}
