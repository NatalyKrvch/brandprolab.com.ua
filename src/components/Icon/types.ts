export interface IconProps {
  size: 's' | 'm' | 'l';
  iconURL: string;
  isSemiTransparent?: boolean;
  iconAlt?: string;
  circleColor?: string;
  isIconCentered?: boolean;
}
