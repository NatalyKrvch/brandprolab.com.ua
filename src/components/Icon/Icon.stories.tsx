import type { Meta, StoryObj } from '@storybook/react';

import { IconClassType } from '@/styles/constants';

import Icon from './Icon';

const meta: Meta<typeof Icon> = {
  title: 'Components/Icon',
  component: Icon,
  parameters: {
    controls: { expanded: true },
  },
  argTypes: {
    iconURL: {
      control: 'text',
      description: 'URL of the icon image (SVG or raster).',
      table: { type: { summary: 'string' } },
    },
    iconAlt: {
      control: 'text',
      description: 'Alt text for the image.',
      table: { type: { summary: 'string' } },
    },
    circleColor: {
      control: 'text',
      description: 'Tailwind CSS class for circle background.',
      table: { type: { summary: 'string' } },
    },
    type: {
      control: { type: 'radio' },
      options: [IconClassType.SERVICES, IconClassType.EXPERTISE],
      description: 'Predefined size and offset classes.',
      table: { type: { summary: 'SERVICES | EXPERTISE' } },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Icon>;

export const ServicesIcon: Story = {
  args: {
    iconURL: '/icons/cards/check.svg',
    iconAlt: 'Check icon',
    circleColor: 'bg-teal-darkOpacity',
    type: IconClassType.SERVICES,
  },
};

export const ExpertiseIcon: Story = {
  args: {
    iconURL: '/icons/cards/check.svg',
    iconAlt: 'Check icon',
    circleColor: 'bg-teal-light',
    type: IconClassType.EXPERTISE,
  },
};
