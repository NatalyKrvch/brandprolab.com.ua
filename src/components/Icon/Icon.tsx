import Image from 'next/image';

import {
  ICON_COMPONENT_IMAGE_TEST_ID,
  ICON_COMPONENT_TEST_ID,
} from '@/lib/testIDs';

import { getIconClasses } from './helpers/getIconClasses';
import { useIconOffset } from './hooks/useIconOffset';
import { IconProps } from './types';

const Icon = ({
  iconURL,
  iconAlt = 'Icon',
  circleColor = 'bg-teal-darkOpacity',
  type,
}: IconProps) => {
  const { circleClass, iconClass } = getIconClasses(type);
  const offset = useIconOffset(type);

  return (
    <div
      className="flex items-center justify-center"
      data-testid={ICON_COMPONENT_TEST_ID}
    >
      <div className={`${circleColor} ${circleClass}`}>
        <Image
          src={iconURL}
          alt={iconAlt}
          width={0}
          height={0}
          className={iconClass}
          style={{
            transform: `translateX(-${offset}px)`,
          }}
          data-testid={ICON_COMPONENT_IMAGE_TEST_ID}
        />
      </div>
    </div>
  );
};

export default Icon;
