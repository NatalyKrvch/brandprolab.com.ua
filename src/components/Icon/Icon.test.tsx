import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import {
  ICON_COMPONENT_IMAGE_TEST_ID,
  ICON_COMPONENT_TEST_ID,
} from '@/lib/testIDs';

import Icon from './Icon';
import { IconClassType } from './types';

jest.mock('next/image', () => ({
  __esModule: true,
  default: ({
    src,
    alt,
    className,
    'data-testid': dataTestId,
  }: {
    src: string;
    alt: string;
    className?: string;
    'data-testid'?: string;
  }) => (
    <img src={src} alt={alt} className={className} data-testid={dataTestId} />
  ),
}));

jest.mock('@/hooks/useMediaQuery', () => ({
  useMediaQuery: (query: string): boolean => {
    if (query.includes('min-width: 1224px')) return true;
    if (query.includes('min-width: 600px')) return false;
    return false;
  },
}));

describe('Icon component', () => {
  const defaultProps = {
    iconURL: '/test-icon.svg',
    iconAlt: 'Test Icon',
    circleColor: 'bg-teal-darkOpacity',
    type: IconClassType.SERVICES,
  };

  it('renders the Icon component with image inside and correct props', () => {
    const { asFragment } = render(<Icon {...defaultProps} />);

    const wrapper = screen.getByTestId(ICON_COMPONENT_TEST_ID);
    const image = screen.getByTestId(ICON_COMPONENT_IMAGE_TEST_ID);

    expect(wrapper).toBeInTheDocument();
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', '/test-icon.svg');
    expect(image).toHaveAttribute('alt', 'Test Icon');
    expect(asFragment()).toMatchSnapshot();
  });
});
