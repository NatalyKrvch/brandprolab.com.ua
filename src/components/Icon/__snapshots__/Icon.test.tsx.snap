// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component renders the Icon component with image inside and correct props 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center"
    data-testid="icon-component-test-id"
  >
    <div
      class="bg-teal-darkOpacity relative flex items-center justify-center rounded-full w-[68px] h-[68px] tablet:w-[80px] tablet:h-[80px] desktop:w-[108px] desktop:h-[108px]"
    >
      <img
        alt="Test Icon"
        class="absolute w-[64px] h-[64px] tablet:w-[72px] tablet:h-[72px] desktop:w-[96px] desktop:h-[96px]"
        data-testid="icon-component-image-test-id"
        src="/test-icon.svg"
        style="transform: translateX(-36px);"
      />
    </div>
  </div>
</DocumentFragment>
`;
